### PR TITLE
chore: remove prop-types

### PR DESCRIPTION
**Pull request by TheSisb**
Removes all prop-types in every component. Marks these changes as major - breaking, with upgrade guide updated.

Also updates `pin-external-deps` danger script to display which package has the unpinned deps.

The danger check is a false positive.

### CODE REVIEW

File: packages/paste-core/core-bundle/progress-bar/package.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (nkrantz) [inline packages/paste-core/core-bundle/progress-bar/package.json:1]: sneaky rebase

Comment (TheSisb) [reply packages/paste-core/core-bundle/progress-bar/package.json:1]: Doh

**Review by SiTaggart — APPROVED** [review]

File: tools/plop-templates/package.hbs [hunk 2]
```diff
@@ -46,7 +46,7 @@
     "@twilio-paste/styling-library": "^2.0.2",
     "@twilio-paste/theme": "^10.0.2",
     "@twilio-paste/types": "^5.0.0",
-    "prop-types": "^15.7.2",
+    ,
```
Comment (SiTaggart) [inline tools/plop-templates/package.hbs:49]: issue: looks like a whoopsie

**Review by SiTaggart — CHANGES_REQUESTED** [review]

File: tools/plop-templates/package.hbs (hunk 2 above)
Comment (TheSisb) [reply tools/plop-templates/package.hbs]: fixed, good catch.

**Pull request merged**